src/storage/src/hummock/iterator/forward_user.rs

```diff
@@ -96,62 +96,20 @@ impl<I: HummockIterator<Direction = Forward>> UserIterator<I> {
     }
 
     /// Gets the iterator move to the next step.
+    /// See `IteratorState` for the state machine details.
     ///
     /// Returned result:
     /// - if `Ok(())` is returned, it means that the iterator successfully move to the next position
     ///   (may reach to the end and thus not valid)
     /// - if `Err(_) ` is returned, it means that some error happened.
     pub async fn next(&mut self) -> HummockResult<()> {
-        while self.iterator.is_valid() {
-            let full_key = self.iterator.key();
-            let epoch = full_key.epoch_with_gap.pure_epoch();
+        // Reset the valid flag to make sure if error happens, `is_valid` should return false.
+        self.is_current_pos_valid = false;
```

Comment:
Seems unnecessary to set it here. `try_advance_to_next_valid` shall be assumed to set the flag to correctly before it returns. In `seek` and `rewind` it is necessary to set the flag before calling `try_advance_to_next_valid` because we have some fast path to early return and not call the `try_advance_to_next_valid`.